vectorstores/cloudsql/vectorstore_container_test.go

```diff
@@ -0,0 +1,232 @@
+// nolint
+package cloudsql_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	tcpostgres "github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/llms/openai"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/util/cloudsqlutil"
+	"github.com/tmc/langchaingo/vectorstores/cloudsql"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+func preCheckEnvSetting(t *testing.T) string {
+	t.Helper()
+
+	if openaiKey := os.Getenv("OPENAI_API_KEY"); openaiKey == "" {
+		t.Skip("OPENAI_API_KEY not set")
+	}
+
+	pgvectorURL := os.Getenv("PGVECTOR_CONNECTION_STRING")
+	if pgvectorURL == "" {
+		pgVectorContainer, err := tcpostgres.RunContainer(
+			context.Background(),
+			testcontainers.WithImage("docker.io/pgvector/pgvector:pg16"),
+			tcpostgres.WithDatabase("db_test"),
+			tcpostgres.WithUsername("user"),
+			tcpostgres.WithPassword("passw0rd!"),
+			testcontainers.WithWaitStrategy(
+				wait.ForLog("database system is ready to accept connections").
+					WithOccurrence(2).
+					WithStartupTimeout(30*time.Second)),
+		)
+		if err != nil && strings.Contains(err.Error(), "Cannot connect to the Docker daemon") {
+			t.Skip("Docker not available")
+		}
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, pgVectorContainer.Terminate(context.Background()))
+		})
+
+		str, err := pgVectorContainer.ConnectionString(context.Background(), "sslmode=disable")
+		require.NoError(t, err)
+
+		pgvectorURL = str
+	}
+
+	return pgvectorURL
+}
+
+func setEngineWithImage(t *testing.T) cloudsqlutil.PostgresEngine {
+	t.Helper()
+	pgvectorURL := preCheckEnvSetting(t)
+	ctx := context.Background()
+	myPool, err := pgxpool.New(ctx, pgvectorURL)
+	if err != nil {
+		t.Fatal("Could not set Engine: ", err)
+	}
+	// Call NewPostgresEngine to initialize the database connection
+	pgEngine, err := cloudsqlutil.NewPostgresEngine(ctx,
+		cloudsqlutil.WithPool(myPool),
+	)
+	if err != nil {
+		t.Fatal("Could not set Engine: ", err)
+	}
+
+	return pgEngine
+}
+
+func initVectorStore(t *testing.T) (cloudsql.VectorStore, func() error) {
```

Comment:
This can not be run multiple times with the same name. When runnign in parallel this table will be duplicated or deleted before the testing is finished